src/hash.rs

```diff
@@ -59,8 +60,17 @@ impl StarkFelt {
         Err(StarknetApiError::OutOfRange { string: hex_str_from_bytes::<32, true>(bytes) })
     }
 
+    /// Returns a new unckecked [`StarkFelt`].
+    ///
+    /// # Safety
+    /// The most significant byte nible of the array must be 0 (`byte[0] < 0x10`).
+    /// Otherwise the resulting [`StarkFelt`] will end up containing an out of range value.
+    pub const unsafe fn new_unchecked(bytes: [u8; 32]) -> Self {
+        Self(bytes)
+    }
+
     /// Storage efficient serialization for field elements.
-    pub fn serialize(&self, res: &mut impl std::io::Write) -> Result<(), Error> {
+    pub fn serialize(&self, res: &mut Vec<u8>) {
```

Comment:
I had to remove anything io dependent.
Do you want me to add back a serialize_from_reader and a deserialize_to_writer function?

Reply:
yes please add them back behind a std feature flag

Reply:
done